toolbox/util/githubClient.go

```diff
@@ -96,9 +96,76 @@ func (g GithubClient) CreatePullRequest(
 	pr, _, err := g.client.PullRequests.Create(
 		context.Background(), g.owner, repo, &req)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	log.Printf("Created new PR at %s", *pr.HTMLURL)
+	return pr, nil
+}
+
+// MergePullRequestOnceCIGreen polls CI status of a PR periodically
+// if success, merge the PR without review
+// if failure, close the PR and its branch
+// if CI still pending after timeout, return error
+func (g GithubClient) MergePullRequestOnceCIGreen(repo string, pr *github.PullRequest) error {
+	prName := fmt.Sprintf("%s/%s#%d", g.owner, repo, pr.GetNumber())
+	timeoutMin := time.Minute * 30
+	intervalMin := time.Minute * 3
+	log.Printf("Waiting for CI results of PR %s\n", prName)
+	log.Printf("Time out in %s, polls every %s\n", timeoutMin, intervalMin)
+	timeStarted := time.Now()
+	for {
+		timeElapsedMin := time.Since(timeStarted)
+		log.Printf("Polling. time elapsed: %s\n", timeElapsedMin)
+		if timeElapsedMin.Nanoseconds() > timeoutMin.Nanoseconds() {
+			return fmt.Errorf("failed to get CI results before timeout")
+		}
+		combinedStatus, _, err := g.client.Repositories.GetCombinedStatus(
+			context.Background(), g.owner, repo, *pr.Head.SHA, nil)
+		if err != nil {
+			return err
+		}
+		switch GetCIState(combinedStatus, nil) {
```

Comment:
Can you rename GetCIState, to get GetRequiredCheckStatuses, and use this API to find them https://developer.github.com/v3/repos/branches/#get-required-status-checks-of-protected-branch

Reply:
The API you suggested remains in perview stage and using postman to ping it has not shown what we would like. Instead, as @yutongz suggests, since the update_deps job runs every 12 hours, any PR that is still open after 12 hours (aka when the job runs the next time) pretty much fails the presubmit and was not merged. We could safely close them in this case.